### PR TITLE
Cleanup CI jobs

### PR DESCRIPTION
**Pull request by osinstom**
This PR:
- splits Docker build CI step into a single step per component
- enables skipping `Push Docker image` step if a PR comes from a fork, to avoid failing the entire CI job.
- adds a configuration file for REUSE linter
- moves licenses into a separate directory as required by REUSE linter
- refactors Dockerfile; `make pb` will not require the BESS stage to be built before. It speeds up generating protobuf files.

### CODE REVIEW

**Comment by osinstom** [comment]
@krsna1729 can you take a look at this PR? Please confirm I've used a correct license for BESS-related files.

**Comment by krsna1729** [comment]
The bess_pb folder is auto-generated.

For anything else bess related BSD-3 is what they say.
https://github.com/NetSys/bess/blob/master/COPYING

File: telemetry.sh [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2021 Open Networking Foundation
```
Comment (krsna1729) [inline telemetry.sh:4]: This is from Intel

File: telemetry.sh [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
```
Comment (pudelkoM) [inline telemetry.sh:2]: remove blank line

**Review by pudelkoM — APPROVED** [review]

**Pull request merged**